### PR TITLE
Refactoring API

### PR DESCRIPTION
**Pull request by fpagnoux**
Some minor refactors in the api : 
- Renaming
- Make anual income assumption explicit
- Merge buildRequest and mapSituation to buildOpenFiscaRequest

### CODE REVIEW

File: lib/simulation/openfisca/index.js [hunk 1]
```diff
@@ -2,31 +2,15 @@
...
-        }],
-        variables: _.keys(PRESTATIONS)
-    };
-}
 
 function calculate(situation, debug, callback) {
     request
         .post(config.openfiscaApi + '/api/1/calculate')
-        .send(buildRequest(situation, debug))
+        .send (mapping.buildOpenFiscaRequest(situation, debug))
```
Comment (MattiSG) [inline lib/simulation/openfisca/index.js]: Space.

File: lib/simulation/openfisca/mapping/index.js [hunk 2]
```diff
@@ -108,6 +105,14 @@ var applyRessources = function(src, openFiscaEntity, schema, dateDeValeur) {
                 _.forEach(agr, function (value, agregatKey) {
                     openFiscaEntity[openFiscaKey][periods[agregatKey].key] = value;
                 });
+
+                // On suppose que les revenus pour l'année fiscale de référence sont les mêmes que pour les 12 derniers mois, si ces revenus n'ont pas encore
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/index.js]: :+1:

File: lib/simulation/openfisca/mapping/index.js [hunk 3]
```diff
@@ -203,17 +210,27 @@ var mapFamille = function(situation) {
...
 };
 
-exports.mapSituation = function(situation) {
-    return {
-        familles: [mapFamille(situation)],
-        foyers_fiscaux: [mapFoyerFiscal(situation)],
-        individus: mapIndividus(situation),
-        menages: [transform(situation, men, situation)]
+exports.buildOpenFiscaRequest = function(situation, debug) {
+        return {
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/index.js]: Indent.

File: lib/simulation/openfisca/mapping/index.js [hunk 4]
```diff
@@ -203,17 +210,27 @@ var mapFamille = function(situation) {
...
+        intermediate_variables: debug,
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/index.js]: It seems pretty annoying to pass this `debug` param over and over. Might be worth looking into whether we can replace it with `process.env.DEBUG` or `process.env.NODE_ENV == 'dev'` someday.

File: lib/simulation/openfisca/mapping/index.js [hunk 5]
```diff
@@ -203,17 +210,27 @@ var mapFamille = function(situation) {
...
+        labels: true,
+        scenarios: [{
+            test_case: {
+                familles: [mapFamille(situation)],
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/index.js]: Add spaces inside inline arrays and objects to improve readability.

File: lib/simulation/openfisca/mapping/periodTools.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 var moment = require('moment');
 
 
-exports.map = function(date) {
+exports.toOpenFiscaFormat = function(date) {
```
Comment (MattiSG) [inline lib/simulation/openfisca/mapping/periodTools.js]: :+1:

**Comment by MattiSG** [comment]
Great job, GTM!

**Comment by fpagnoux** [comment]
@MattiSG 
FI, I did more renaming. As you are away and all the tests are passing, I'll merge, but let me know if you have additional feedback.

**Pull request merged**